src/schemes/fluid/viscosity.jl

```diff
@@ -118,10 +164,22 @@ end
     # viscosity is used for shocks and not rarefactions."
     if vr < 0
         mu = h * vr / (distance^2 + epsilon * h^2)
-        return -(alpha * c * mu + beta * mu^2) / rho_mean
+        return -(alpha * c * mu + beta * mu^2) / rho_mean * grad_kernel
     end
 
-    return 0.0
+    return zero(v_diff)
+end
+
+@inline function (viscosity::ViscosityMorris)(c, v_diff, vr, distance, rho_mean, rho_a,
+                                              rho_b, h, grad_kernel)
+    (; epsilon, nu) = viscosity
+
+    # TODO This is not correct for two different fluids. It should be `nu_a` and `nu_b`.
+    mu_a = nu * rho_a
+    mu_b = nu * rho_b
+
+    return (mu_a + mu_b) / (rho_a * rho_b) * dot(v_diff, grad_kernel) /
+           (distance + epsilon * h)
```

Comment:
```suggestion
    return (mu_a + mu_b) / (rho_a * rho_b) * dot(pos_diff, grad_kernel) /
           (distance + epsilon * h) * v_diff
```
Otherwise, this is a scalar. We also need unit tests that can at least catch something like this.